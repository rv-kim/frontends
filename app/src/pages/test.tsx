import * as React from 'react'
import { connect } from 'react-redux'
import Link from 'next/link'
import axios from 'axios'

export class Test extends React.Component {
  // static async getInitialProps ({ store }) {
  //   const res = await axios.get('http://demo2269115.mockable.io/test')
  //   return {text: res.data.msg}
  // }

  render () {
    // const { text } = this.props
    const text = ".."
    return (
      <div>
        <Link href="/index"><a>link to top {text}</a></Link>
        <hr/>
      </div>
    )
  }
}

export default connect()(Test)
