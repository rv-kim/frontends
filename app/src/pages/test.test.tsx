import * as React from 'react'
import { shallow } from 'enzyme'
import * as ShallowRenderer from 'react-test-renderer/shallow'
// import configureStore from 'redux-mock-store'
import { Test } from './test'

const renderer = new ShallowRenderer();
// const mockStore = configureStore()

describe('Test Component', () => {

  test('snapshots_01', () => {
    const ss = renderer.render(<Test />);
    expect(ss).toMatchSnapshot()
  })

  test('test 01', () => {
    // const store = mockStore({count: 0})
    // const test = shallow(<Test store={store} />)
    const test = shallow(<Test />)
    expect(test).toEqual(test)
  })
})
