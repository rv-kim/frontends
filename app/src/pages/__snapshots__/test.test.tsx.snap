// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component snapshots_01 1`] = `
<div>
  <Link
    href="/index"
  >
    <a>
      link to top 
      ..
    </a>
  </Link>
  <hr />
</div>
`;
